crates/hotshot/task-impls/src/transactions.rs

```diff
@@ -366,7 +360,50 @@ impl<TYPES: NodeType, V: Versions> TransactionTaskState<TYPES, V> {
                     .leader(view)
                     .await?;
                 if leader == self.public_key {
-                    self.handle_view_change(&event_stream, view, *epoch).await;
+                    self.handle_view_change(&event_stream, view, *epoch, None)
+                        .await;
+                    return Ok(());
+                }
+            },
+            HotShotEvent::QuorumProposalValidated(proposal, _leaf) => {
+                let view_number = proposal.data.view_number();
+                let next_view = view_number + 1;
+
+                let version = match self.upgrade_lock.version(next_view).await {
+                    Ok(v) => v,
+                    Err(e) => {
+                        tracing::error!("Failed to calculate version: {e:?}");
+                        return Ok(());
+                    },
+                };
+
+                if version < V::DrbAndHeaderUpgrade::VERSION {
+                    return Ok(());
+                }
+
+                let vid = proposal.data.block_header().payload_commitment();
+                let block_height = proposal.data.block_header().block_number();
+                if is_epoch_transition(block_height, self.epoch_height) {
+                    return Ok(());
+                }
+                if is_last_block(block_height, self.epoch_height) {
+                    return Ok(());
+                }
```

Comment:
nit: I think this is unnecessary as `is_last_block` is already covered by `is_epoch_transition`.